quickstart/101-sql-database/main.tf

```diff
@@ -0,0 +1,44 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+resource "random_string" "azurerm_mssql_server_name" {
+  length  = 13
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "random_password" "admin_password" {
+  count       = var.admin_password == null ? 1 : 0
+  length      = 20
+  special     = true
+  min_numeric = 1
+  min_upper   = 1
+  min_lower   = 1
+  min_special = 1
+}
+
+locals {
+  admin_password = try(random_password.admin_password[0].result, var.admin_password)
+}
+
+resource "azurerm_mssql_server" "server" {
+  name                         = "sql${random_string.azurerm_mssql_server_name.result}"
```

Comment:
This would be cleaner to do this with a random_pet resource:
```terraform
resource "random_pet" "this" {
   prefix        = "sql"
   separator = ""
}

Unless there is a reason to go with random_string instead.

Reply:
I'm fine with this but I believe we should allow the default separator as it is [allowed in sql server names](https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.SQL.ServerName/#:~:text=The%20requirements%20for%20SQL%20logical%20server%20names%20are%3A,SQL%20logical%20server%20names%20must%20be%20globally%20unique.) and it makes the name more readable.